pkg/client/gen.go

```diff
@@ -119,6 +119,36 @@ func (*SonobuoyClient) GenerateManifest(cfg *GenConfig) ([]byte, error) {
 		return nil, errors.Wrap(err, "plugin YAML generation")
 	}
 
+	if cfg.PluginEnvOverrides != nil {
+		for pluginName, envVars := range cfg.PluginEnvOverrides {
+			found := false
+			for _, p := range plugins {
+				if p.SonobuoyConfig.PluginName == pluginName {
```

Comment:
Maybe throw an error if a plugin doesn't match?

Reply:
See just a few lines farther down: https://github.com/heptio/sonobuoy/pull/709/files#diff-bbcb11cee7bcaf5cb4cf6e2cd7dc61bbR147 Since the plugins are a slice instead of a map I can't just do a quick lookup; so I just track whether it was found in the list of plugins or not.

I try to give a really helpful message to make things like typos clear (so they can see, for instance, they typed `system-logs` instead of `systemd-logs`)